### DISCUSSION TITLE
Querying Index using multiple dynamic fields not working

### DISCUSSION THREAD

**Discussion by Sindraksi** [ravendb/ravendb#17144] [Q&A]
Hi,

I am using Raven 5.3 and can't get my search index working with mutliple dynamic fields (using the CreateField() method when creating the index). 
Here is the code used to create the indexed dynamic fields:

_ = this0.attribute.Values.Select(x0 => this.CreateField(x0.Code, x0.Value, false, true))

When the first dynamic field is evaluated, all other dynamic fields are no longer recognised. E.g

Query 1
```
from index 'IndexName'
where search("dynamicField1", '*') 
```

will return 100 reults as expexted

Query 2
```
from index 'IndexName'
where search("dynamicField2", '*') 
```

will return 100 results as expexted

Query 3
```
from index 'IndexName'
where search("dynamicField1", '*') and search("dynamicField2" = '*')
```

will return nothing, but the expectation would be 100 results the same since both search terms are wildcarded.

This also causes issues with to Facets:

```
facetName
{
    "Facets": [
        {
            "FieldName": "dynamicField1",
            "Options": null,
            "Aggregations": {},
            "DisplayFieldName": null
        },
        {
            "FieldName": "dynamicField2",
            "Options": null,
            "Aggregations": {},
            "DisplayFieldName": null
        },
        {
            "FieldName": "dynamicField3",
            "Options": null,
            "Aggregations": {},
            "DisplayFieldName": null
        },
        {
            "FieldName": "dynamicField4",
            "Options": null,
            "Aggregations": {},
            "DisplayFieldName": null
        },
        {
            "FieldName": "staticField1",
            "Options": null,
            "Aggregations": {},
            "DisplayFieldName": null
        }
    ],
    "RangeFacets": [],
    "@metadata": {
        "@collection": "FacetSetups",
        "Raven-Clr-Type": "Raven.Client.Documents.Queries.Facets.FacetSetup, Raven.Client"
    }
}
```

```
from index 'IndexName'
where search("dynamicField1", '*')
select facet(id("facetName"))
```
the results will only have values for dynamicField1 and staticField1
dynamicField2, dynamicField3 and dynamicField4 will have completely empty values

The final issue this is causing is order by: 

```
from index 'IndexName'
where search("dynamicField1", '*')
order by 'dynamicField2' as alphanumeric
```
in the above example the order by does not do anything as dynamicField2 is not recognised once dynamicField1 has been used.

**Comment by garayx** [comment]
Hi

RavenDB 5.3 is out of support, can you try running latest RavenDB 5.4 stable (https://ravendb.net/download) and see if the code works ?

In case it doesn't please send a unit-test (https://ravendb.net/docs/article-page/5.4/csharp/start/test-driver#example-1) or simple app with reproduce

**Reply by Sindraksi** [reply]
Thanks, I have tried upgrading to 5.4 but still had these issues. I have not had a chance to set up any unit tests or app, I will get back to you when I can.
